specreduce/core.py

```diff
@@ -2,3 +2,246 @@
+
+    if std is None:
+        msg1 = "Error: No 'onedstd' standard star file has been specified." \
+               "Cannot proceed further."
+        raise ValueError(msg1)
+
+    std = std.lower()
+    std_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "datasets", "onedstds")
+
+    if os.path.isfile(os.path.join(std_dir, std)):
+        std_wave, std_mag, std_wth = np.genfromtxt(
+            os.path.join(std_dir, std), skip_header=1, unpack=True
+        )
+
+        # Convert standard star from magnitude units to flux
+        std_flux = _mag2flux(std_wave, std_mag)
+
+        # Define an array of the three most prominent Balmer lines
+        balmer = np.array([6563, 4861, 4341], dtype="float")
+
+        # Down-sample (coded as ds) the observed flux to fit the standard star's bins
+        obj_flux_ds = []
+        obj_wave_ds = []
+        std_flux_ds = []
+
+        for i in range(len(std_wave)):
+            rng = np.where((obj_wave >= std_wave[i] - std_wth[i] / 2.0) &
+                           (obj_wave < std_wave[i] + std_wth[i] / 2.0))
+            isbalmer = np.where((balmer >= std_wave[i] - std_wth[i] / 2.0) &
+                                (balmer < std_wave[i] + std_wth[i] / 2.0))
+
+            # Check to see if the bin does indeed contain observed spectra,
+            # with no presence of the Balmer lines defined previously
+            if (len(rng[0]) > 1) and (len(isbalmer[0]) == 0):
+                obj_flux_ds.append(np.nanmean(obj_flux[rng]))
+                obj_wave_ds.append(std_wave[i])
+                std_flux_ds.append(std_flux[i])
+
+        # To ensure the ratio between the standard star flux and observed flux
+        # has proper flux units
+        ratio = np.abs(np.array(std_flux_ds, dtype="float") /
+                       np.array(obj_flux_ds, dtype="float"))
+
+        # Interpolate flux calibration (sensfunc) onto object's wavelength grid
+        # 3 modes of interpolation available: 'linear', 'cubic spline', and 'polynomial'
+
+        # To check if a mode of interpolation has been specified
+        if mode not in ("linear", "spline", "poly"):
+            mode = "spline"
+            print("Warning: No interpolation mode has been set"
+                  "Setting mode to the default 'spline'.")
+
+        # Fit the log of the above sensfunc ratio
+        # As IRAF uses 2.5*log(ratio), the quantities are in mag units
+        logsensfunc = np.log10(ratio)
+
+        # Interpolate the logsensfunc back onto the wavelength grid
+        if mode == "linear":
+            sensfunc = np.interp(obj_wave, obj_wave_ds, logsensfunc)
+        elif mode == "spline":
+            spline = UnivariateSpline(obj_wave_ds, logsensfunc, ext=0, k=2, s=0.0025)
+            sensfunc = spline(obj_wave)
+        elif mode == "poly":
+            fit = np.polyfit(obj_wave_ds, logsensfunc, polyorder)
+            sensfunc = np.polyval(fit, obj_wave)
+
+        if show is True:
```

Comment:
a better pattern for providing graphical output is to bundle the plot into a `matplotlib.figure.Figure` object that is then returned to the caller. how figures are handled will vary depending on context (interactive vs non-interactive, which backend is used, etc) so we don't want to bake assumptions into core code here.

Reply:
Agreed

Reply:
newbie Q: do we have a demo for how to do this properly?